src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -153,7 +153,7 @@ internal JsonTypeInfo(Type type, JsonConverter converter, JsonSerializerOptions
             {
                 VerifyMutable();
 
-                if (Kind != JsonTypeInfoKind.Object)
+                if (Kind == JsonTypeInfoKind.None)
```

Comment:
In the unlikely case that new kinds get added in the future, I would probably just turn this into an explicit check:
```suggestion
                if (Kind is not (JsonTypeInfoKind.Object or JsonTypeInfoKind.Enumerable or JsonTypeInfoKind.Dictionary))
```

Reply:
Nice precaution.